Model/SentryPerformance.php

```diff
@@ -0,0 +1,153 @@
+        private Data $helper
+    ) {
+    }
+
+    public function startTransaction(AppInterface $app): void
+    {
+        if (!$app instanceof Http) {
+            // actually, we only support profiling of http requests.
+            return;
+        }
+
+        $requestStartTime = $this->request->getServer('REQUEST_TIME_FLOAT', microtime(true));
+
+        $context = TransactionContext::fromHeaders(
+            $this->request->getHeader('sentry-trace') ?: '',
+            $this->request->getHeader('baggage') ?: ''
+        );
+
+        $requestPath = '/'.ltrim($this->request->getRequestUri(), '/');
+
+        $context->setName($requestPath);
+        $context->setSource(TransactionSource::url());
+        $context->setStartTimestamp($requestStartTime);
+
+        $context->setData([
+            'url'    => $requestPath,
+            'method' => strtoupper($this->request->getMethod()),
+        ]);
+
+        // Start the transaction
+        $transaction = startTransaction($context);
+
+        // If this transaction is not sampled, don't set it either and stop doing work from this point on
+        if (!$transaction->getSampled()) {
+            return;
+        }
+
+        $this->transaction = $transaction;
+        SentrySdk::getCurrentHub()->setSpan($transaction);
+    }
+
+    public function finishTransaction(ResponseInterface|int $statusCode): void
+    {
+        if ($this->transaction === null) {
+            return;
+        }
+
+        try {
+            $state = $this->objectManager->get(State::class);
+            $areaCode = $state->getAreaCode();
+        } catch (LocalizedException) {
+            // we wont track transaction without an area
+            return;
+        }
+
+        if (in_array($areaCode, $this->helper->getPerformanceTrackingExcludedAreas())) {
+            return;
+        }
+
+        if ($statusCode instanceof Response) {
+            $statusCode = (int) $statusCode->getStatusCode();
+        }
+
+        if (is_numeric($statusCode)) {
+            $this->transaction->setHttpStatus($statusCode);
+        }
+
+        if (in_array($state->getAreaCode(), ['frontend', 'webapi_rest', 'adminhtml'])) {
+            if (!empty($this->request->getFullActionName())) {
+                $this->transaction->setName(strtoupper($this->request->getMethod()). ' ' .$this->request->getFullActionName());
+            }
+
+            $this->transaction->setOp('http');
+
+            $this->transaction->setData(array_merge(
+                $this->transaction->getData(),
+                $this->request->__debugInfo(),
+                [
+                    'module' => $this->request->getModuleName(),
+                    'action' => $this->request->getFullActionName(),
+                ]
+            ));
+        } elseif ($state->getAreaCode() === 'graphql') {
+            $this->transaction->setOp('graphql');
+        } else {
+            $this->transaction->setOp($state->getAreaCode());
+        }
```

Comment:
Was there anything special that needed to happen here? If not we can remove the elseif
```suggestion
       } else {
            $this->transaction->setOp($state->getAreaCode());
        }
```